src/Utilities/LockCheck.cs

```diff
@@ -14,8 +14,13 @@
 
 namespace Microsoft.Build.Utilities
 {
+    /// <summary>
+    /// This class implements checking what processes are locking a file on Windows.
+    /// It uses the Restart Manager API to do this.
+    /// Use the method <see cref="GetLockedFileMessage"/> to get a message to inform the user which processes have a lock on a given file.
+    /// </summary>
     [SupportedOSPlatform("windows")]
```

Comment:
Having the whole class only supported on Windows pushes the responsibility onto callers to have a calling pattern like:

1. [do some file operation that fails]
2. Check if on Windows and if so call `LockCheck`
3. Accumulate the returned string into an error

Could/should we make that easier for the user? Or is that the right level of abstraction to require?

Reply:
The windows check makes more sense in the LockCheck class imo, so I refactored that. 
I would characterize it like this now:
1. [file op fails]
2. try getting a part of message `m` by calling LockCheck to get what is locking the file if applicable (it can be nothing or not supported or not windows -> string.Empty)
3. accumulate `m` to error
(this is the same functionality as before) lmk if it's better/enough or whether I should think harder about refactoring